### PR TITLE
Add complete parameter description documentation

### PR DESCRIPTION
**Pull request by SteveMacenski**
Adding complete descriptions for all parameters in the stack.

Currently missing 
- [x] AMCL
- [x] DWB

Need to review some notes to add those, but the rest are here

### CODE REVIEW

**Comment by SteveMacenski** [comment]
@jackpien definitely not looking to over-work you, but if you're sufficiently interested, adding the parameter descriptions to DWB would be massively helpful. I can do AMCL next week, I need some notes I took a few years ago for it.

**Comment by SteveMacenski** [comment]
@Marwan99 can you take a look over this?

**Comment by Marwan99** [comment]
sure

File: doc/parameters/param_list.md [hunk 1]
```diff
@@ -1,99 +1,98 @@
...
-| transform_tolerance | 0.1 |  |
-| global_frame | "map" |  |
-| robot_base_frame | "base_link" |  |
+| bt_xml_filename | N/A | path to behavior tree XML description |
+| plugin_lib_names | ["nav2_compute_path_to_pose_action_bt_node", "nav2_follow_path_action_bt_node", "nav2_back_up_action_bt_node", "nav2_spin_action_bt_node", "nav2_wait_action_bt_node", "nav2_clear_costmap_service_bt_node", "nav2_is_stuck_condition_bt_node", "nav2_goal_reached_condition_bt_node", "nav2_initial_pose_received_condition_bt_node", "nav2_goal_updated_condition_bt_node", "nav2_reinitialize_global_localization_service_bt_node", "nav2_rate_controller_bt_node", "nav2_distance_controller_bt_node", "nav2_recovery_node_bt_node", "nav2_pipeline_sequence_bt_node", "nav2_round_robin_node_bt_node", "nav2_transform_available_condition_bt_node"] | list of behavior tree node shared library |
```
Comment (Marwan99) [inline doc/parameters/param_list.md:8]: Replace library with libraries?

File: doc/parameters/param_list.md [hunk 2]
```diff
@@ -102,328 +101,332 @@ Namespace: /parent_ns/local_ns
...
 | Parameter | Default | Description |
 | ----------| --------| ------------|
-| `<dwb plugin>`.`<name>`.oscillation_reset_dist | 0.05 | |
-| `<dwb plugin>`.`<name>`.oscillation_reset_angle | 0.2 | |
-| `<dwb plugin>`.`<name>`.oscillation_reset_time | -1 | |
-| `<dwb plugin>`.`<name>`.x_only_threshold | 0.05 | |
-| `<dwb plugin>`.`<name>`.scale | 1.0 | |
+| `<dwb plugin>`.`<name>`.oscillation_reset_dist | 0.05 | Minimum distance to move to reset oscillation watchdog (m) |
+| `<dwb plugin>`.`<name>`.oscillation_reset_angle | 0.2 | Minimum angular distance to move to reset watchdog (rad) |
+| `<dwb plugin>`.`<name>`.oscillation_reset_time | -1 | Duration when a reset may be called |
```
Comment (Marwan99) [inline doc/parameters/param_list.md:188]: What happens when it set to -1?

**Comment by SteveMacenski** [comment]
Thanks for the review, @Marwan99 any other new ones you see? resolved all the ones you mentioned.

**Comment by Marwan99** [comment]
Just spotted BT Node ComputePathToPose `planner_id` and BT Node FollowPath `controller_id` are not populated. Thats everything I can see, thanks.

**Comment by SteveMacenski** [comment]
Ah thanks, thanks for pointing that out.

Waiting on CI then. How's the new example web-page going?

**Comment by Marwan99** [comment]
Having a look around atm to see whats out there, then will create a sample/template and push it for feedback. (ETA 1-2 days)

**Comment by SteveMacenski** [comment]
Awesome :-) thanks for your help in jump starting this. If you can get a couple of the pages done, I’ll commit to doing the rest.

**Comment by jackpien** [comment]
> @jackpien definitely not looking to over-work you, but if you're sufficiently interested, adding the parameter descriptions to DWB would be massively helpful.

Hi @SteveMacenski - I'll push this to do next (after mono-ORB-SLAM2). I know this task isn't much but I'm at max capacity context switching already. If time is of essence and someone else wants to take a stab and have me review it, that'll probably be slightly more efficient.

**Comment by SteveMacenski** [comment]
@jackpien I already did it - in this PR

File: doc/parameters/param_list.md [hunk 3]
```diff
@@ -102,328 +101,332 @@ Namespace: /parent_ns/local_ns
...
 
 * `<name>`: twirling critic name defined in `<dwb plugin>.critics`
 
 | Parameter | Default | Description |
 | ----------| --------| ------------|
-| `<dwb plugin>`.`<name>`.scale | 0.0 | |
+| `<dwb plugin>`.`<name>`.scale | 0.0 | Weight scale |
 
-### goal_align TrajectoryCritic
+## goal_align TrajectoryCritic
```
Comment (jackpien) [inline doc/parameters/param_list.md:257]: GoalAlignCritic inherits from GoalDistCritic which inherits from MapGridCritic. 

If we are including parameters from parent classes, then GoalAlignCritic should also include the aggregation_type parameter (from MapGridCritic) 

| `<dwb plugin>`.`<name>`.aggregation_type | "last" | last, sum, or product combination methods |

File: doc/parameters/param_list.md [hunk 4]
```diff
@@ -102,328 +101,332 @@ Namespace: /parent_ns/local_ns
...
 
 | Parameter | Default | Description |
 | ----------| --------| ------------|
-| `<dwb plugin>`.`<name>`.aggregation_type | "last" | |
-| `<dwb plugin>`.`<name>`.scale | 1.0 | |
+| `<dwb plugin>`.`<name>`.aggregation_type | "last" | last, sum, or product combination methods |
+| `<dwb plugin>`.`<name>`.scale | 1.0 | Weight scale |
 
-### path_align TrajectoryCritic
+## path_align TrajectoryCritic
```
Comment (jackpien) [inline doc/parameters/param_list.md:284]: PathAlignCritic inherits from PathDistCritic which inherits from MapGridCritic. 

If we are including parameters from parent classes, then PathAlignCritic should also include the aggregation_type parameter (from MapGridCritic) 

| `<dwb plugin>`.`<name>`.aggregation_type | "last" | last, sum, or product combination methods |

File: doc/parameters/param_list.md [hunk 5]
```diff
@@ -102,328 +101,332 @@ Namespace: /parent_ns/local_ns
...
 
-### twirling TrajectoryCritic
+## twirling TrajectoryCritic
 
 * `<name>`: twirling critic name defined in `<dwb plugin>.critics`
 
 | Parameter | Default | Description |
 | ----------| --------| ------------|
-| `<dwb plugin>`.`<name>`.scale | 0.0 | |
+| `<dwb plugin>`.`<name>`.scale | 0.0 | Weight scale |
```
Comment (jackpien) [inline doc/parameters/param_list.md:256]: Missing GoalDistCritic?

## goal_dist TrajectoryCritic

| `<dwb plugin>`.`<name>`.aggregation_type | "last" | last, sum, or product combination methods |
| `<dwb plugin>`.`<name>`.scale | 1.0 | Weight scale |

**Comment by jackpien** [comment]
> @jackpien I already did it - in this PR

Thanks @SteveMacenski , Not sure if I'm too late to the game here but added my review comments to the PR.

File: doc/parameters/param_list.md (hunk 5 above)
Comment (SteveMacenski) [reply doc/parameters/param_list.md:256]: its here: https://github.com/ros-planning/navigation2/pull/1786/files#diff-654f280b3f7a18a77bdf3f828c8538f1R166

**Comment by SteveMacenski** [comment]
Its never too late! Thanks for the notes, I made those changes.

**Comment by SteveMacenski** [comment]
@Marwan99 let me know when you're good to merge / for me to review the website PR again. I'll use that as a signal for me to migrate this work over to that website page and then delete this markdown file so we dont have redundant files.

**Pull request merged**